Jenkinsfile

```diff
@@ -19,49 +19,97 @@ pipeline {
       }
     }
 
-    stage('Linting') {
-      steps {
-        sh './bin/check_style'
-
-        checkstyle pattern: 'test/golint.xml', canComputeNew: true, usePreviousBuildAsReference: false, failedNewAll: "0", failedTotalAll: "0",  unHealthy: "0", healthy: "1", thresholdLimit: "low", useDeltaValues: false
-      }
-    }
-
     stage('Run Tests') {
+
       parallel {
+
         stage('Unit tests') {
           steps {
             sh './bin/test_unit'
 
-            junit 'test/junit.xml'
+            junit 'test/unit-test-output/junit.xml'
           }
         }
 
-        stage('Integration tests') {
+        stage('Integration: AWS Secrets Provider') {
           steps {
-            sh './bin/test_integration'
+            sh './bin/run_integration aws_secrets_provider'
+            junit 'test/aws_secrets_provider/junit.xml'
+          }
+        }
 
-            junit 'test/junit.xml'
+        stage('Integration: Conjur') {
+          steps {
+            sh './bin/run_integration conjur'
+            junit 'test/conjur/junit.xml'
           }
         }
 
-        stage('Demo tests') {
+        stage('Integration: HTTP Basic Auth') {
           steps {
-            sh './bin/test_demo'
+            sh './bin/run_integration http_basic_auth'
+            // junit 'test/http_basic_auth/junit.xml'
```

Comment:
It wasn't working, and iirc it wasn't producing output.  Like, it looked like it never was.  I'll take another look and if that's correct, I'll just remove the commented line unless you think otherwise...